plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployOutput.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import java.util.List;
+
+/**
+ * Holds de-serialized JSON output of gcloud app deploy. Don't change the field names
+ * because Gson uses it for automatic de-serialization.
+ */
+// TODO: move into appengine-plugins-core
+// TODO expand to include other Version attributes
+public class AppEngineDeployOutput {
+  private static class Version {
+    String id;
+    String service;
+  }
+
+  private List<Version> versions;
+
+  /**
+   * @return version, can be null
+   */
+  public String getVersion() {
+    if (versions == null || versions.size() != 1) {
```

Comment:
The json reports the version of the deployed project as the version.id, so the id (which is actually the version) could be null